shared/src/TwinCollection.cs

```diff
@@ -176,9 +176,9 @@ public Metadata GetMetadata()
         /// Gets the LastUpdated time for this property
         /// </summary>
         /// <returns>DateTime instance representing the LastUpdated time for this property</returns>
-        public DateTime GetLastUpdated()
+        public DateTime? GetLastUpdated()
         {
-            return (DateTime)_metadata[LastUpdatedName];
+            return (DateTime?)_metadata[LastUpdatedName];
```

Comment:
DateTime isn't a nullable type and we should check for the property here rather than return a null. Changing this to return null could break more people who already depend on this property.

For example, if a customer is expecting a DateTIme object (which is never null), the NRE will be moved to their code.

```
using var deviceClient = DeviceClient.CreateFromConnectionString("<SNIP>");
	
var twin = await deviceClient.GetTwinAsync();

// Assume this normally has a value and we check for it...
var lastUpdated = twin.Properties.Reported.GetLastUpdated();

if (lastUpdate > DateTime.Now.AddHours(-1))
{
    // Do something...
}

```